src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs

```diff
@@ -299,5 +320,60 @@ private void AddEmbeddedStatementsIndentationOperation(List<IndentBlockOperation
                 AddIndentBlockOperation(list, firstToken, lastToken, TextSpan.FromBounds(firstToken.FullSpan.Start, lastToken.FullSpan.End));
             }
         }
+
+        private readonly struct CachedOptions : IEquatable<CachedOptions>
+        {
+            public readonly LabelPositionOptions LabelPositioning;
+            public readonly bool IndentBlock;
+            public readonly bool IndentSwitchCaseSection;
+            public readonly bool IndentSwitchCaseSectionWhenBlock;
+            public readonly bool IndentSwitchSection;
+
+            public CachedOptions(AnalyzerConfigOptions? options)
+            {
+                LabelPositioning = GetOptionOrDefault(options, CSharpFormattingOptions2.LabelPositioning);
+                IndentBlock = GetOptionOrDefault(options, CSharpFormattingOptions2.IndentBlock);
+                IndentSwitchCaseSection = GetOptionOrDefault(options, CSharpFormattingOptions2.IndentSwitchCaseSection);
+                IndentSwitchCaseSectionWhenBlock = GetOptionOrDefault(options, CSharpFormattingOptions2.IndentSwitchCaseSectionWhenBlock);
+                IndentSwitchSection = GetOptionOrDefault(options, CSharpFormattingOptions2.IndentSwitchSection);
+            }
+
+            public static bool operator ==(CachedOptions left, CachedOptions right)
+                => left.Equals(right);
+
+            public static bool operator !=(CachedOptions left, CachedOptions right)
+                => !(left == right);
+
+            private static T GetOptionOrDefault<T>(AnalyzerConfigOptions? options, Option2<T> option)
+            {
+                if (options is null)
+                    return option.DefaultValue;
+
+                return options.GetOption(option);
+            }
+
+            public override bool Equals(object? obj)
+                => obj is CachedOptions options && Equals(options);
+
+            public bool Equals(CachedOptions other)
+            {
+                return LabelPositioning == other.LabelPositioning
+                    && IndentBlock == other.IndentBlock
+                    && IndentSwitchCaseSection == other.IndentSwitchCaseSection
+                    && IndentSwitchCaseSectionWhenBlock == other.IndentSwitchCaseSectionWhenBlock
+                    && IndentSwitchSection == other.IndentSwitchSection;
+            }
+
+            public override int GetHashCode()
+            {
+                var hashCode = 0;
+                hashCode = (hashCode << 2) + (int)LabelPositioning;
+                hashCode = (hashCode << 1) + (IndentBlock ? 1 : 0);
+                hashCode = (hashCode << 1) + (IndentSwitchCaseSection ? 1 : 0);
+                hashCode = (hashCode << 1) + (IndentSwitchCaseSectionWhenBlock ? 1 : 0);
+                hashCode = (hashCode << 1) + (IndentSwitchSection ? 1 : 0);
+                return hashCode;
```

Comment:
no HashCode.Combine?

Reply:
This is faster. Even if a conflict happens the hash will be fast.